skimage/filters/thresholding.py

```diff
@@ -713,3 +715,142 @@ def threshold_triangle(image, nbins=256):
         arg_level = nbins - arg_level - 1
 
     return bin_centers[arg_level]
+
+
+def threshold_multiotsu(image, nclass=3):
+    """Generates multiple thresholds for an input image. Based on the
+    Multi-Otsu approach by Liao, Chen and Chung.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Grayscale input image.
+    nclass : int, optional
+        Number of classes to be thresholded, i.e. the number of resulting
+        regions. Accepts an integer from 2 to 5.
+    nbins : int, optional
+        Number of bins used to calculate the histogram. This value is
+        ignored for integer arrays.
+
+    Returns
+    -------
+    idx_thresh : array
+        Array containing the threshold values for the desired classes.
+
+    References
+    ----------
+    .. [1] Liao, P-S., Chen, T-S. and Chung, P-C., "A fast algorithm for
+    multilevel thresholding", Journal of Information Science and
+    Engineering 17 (5): 713-727, 2001.
+    .. [2] Tosa, Y., "Multi-Otsu Threshold", a java plugin for ImageJ.
+    Available at:
+    http://imagej.net/plugins/download/Multi_OtsuThreshold.java
+
+    Examples
+    --------
+    >>> from skimage.data import camera
+    >>> image = camera()
+    >>> thresh = threshold_multiotsu(image)
+    >>> region1 = image <= thresh[0]
+    >>> region2 = (image > thresh[0]) & (image <= thresh[1])
+    >>> region3 = image > thresh[1]
+    """
+
+    if image.shape[-1] in (3, 4):
+        raise TypeError("The input image seems to be RGB (shape: {0}. Please"
+                        "use a grayscale image.".format(image.shape))
+
+    if image.min() == image.max():
+        raise TypeError("The input image seems to have only one color: {0}."
+                        "Please use a grayscale image.".format(image.min()))
+
+    # check if nclass is between 2 and 5.
+    if nclass not in np.array((2, 3, 4, 5)):
+        raise ValueError("Please choose a number of classes between"
+                         "2 and 5.")
+
+    # image needs to be between [0, 255].
+    nbins = 256
+    image = img_as_ubyte(image)
```

Comment:
I don't like to force nbins and convert the image. It's not necessary. You can get the `dtype_limits` from the image. We have a function in skimage for that.